apps/els_lsp/src/els_sheldon_diagnostics.erl

```diff
@@ -0,0 +1,119 @@
+%%==============================================================================
+%% Compiler diagnostics
+%%==============================================================================
+
+-module(els_sheldon_diagnostics).
+
+%%==============================================================================
+%% Behaviours
+%%==============================================================================
+
+-behaviour(els_diagnostics).
+
+%%==============================================================================
+%% Exports
+%%==============================================================================
+
+-export([ init/0
+        , is_default/0
+        , run/1
+        , source/0
+        ]).
+
+%%==============================================================================
+%% Includes
+%%==============================================================================
+
+-include("els_lsp.hrl").
+-include_lib("kernel/include/logger.hrl").
+
+%%==============================================================================
+%% Callback Functions
+%%==============================================================================
+
+-spec init() -> ok.
+init() ->
+  application:ensure_all_started(sheldon),
```

Comment:
Maybe add a comment that this is executed here because sheldon has a long start time (something we should look into) and therefore we only do it if sheldon diagnostics are enabled?